docs/getting_started.md

```diff
@@ -0,0 +1,206 @@
+  }
+}
+```
+
+This test uses the `Get-AzBPContainerRegistry` helper from BenchPress. `Get-AzBPContainerRegistry` returns an object representing the container registry in the specified resource group and with the specified container registry name. Assuming the container registry exists, we assert that the object returned by `Get-AzBPContainerRegistry` is not null or empty.
+
+Let's look at the second `Describe` block:
+
+```PowerShell
+Describe 'Verify Container Registry Exists' {
+  it 'Should contain a container registry with the given name' {
+    #arrange
+    $rgName = "rg-test"
+    $acrName = "acrbenchpresstest1"
+
+    #act
+    $exists = Get-AzBPContainerRegistryExist -ResourceGroupName $rgName -Name $acrName
+
+    #assert
+    $exists | Should -Be $true
+  }
+}
+```
+
+This test looks very similar to the first test, except it is using the `Get-AzBPContainerRegistryExist` helper instead. This helper doesn't return the object representing the Container Registry, but instead returns true if the Container Registry exists and false if it does not exists. Assuming the container registry exists, we assert that the return should be true.
+
+Now let's look at the third `Describe` block:
+
+```PowerShell
+Describe 'Spin up , Tear down Container Registry' {
+  it 'Should deploy a bicep file.' {
+    #arrange
+    $resourceGroupName = "rg-test"
+    $bicepPath = "./containerRegistry.bicep"
+    $params = @{
+      name           = "acrbenchpresstest2"
+      location       = "westus3"
+    }
+
+    #act
+    $deployment = Deploy-AzBPBicepFeature -BicepPath $bicepPath -Params $params -ResourceGroupName $resourceGroupName
+
+    #assert
+    $deployment.ProvisioningState | Should -Be "Succeeded"
+
+    #clean up
+    Remove-AzBPBicepFeature -ResourceGroupName $resourceGroupName
+  }
+}
+```
+
+This test is unlike the first two tests because we are using the `Deploy-AzBPBicepFeature` helper to deploy a bicep file. `Deploy-AzBicepFeature` will take a bicep file and deploy it to an existing resource group, and it will return the deployment details. As you can see, we instead assert that the deployment should be successful. We then tear down the deployed resources using the `Remove-AzBPBicepFeature` helper.
+
+## Fixing the Tests
+
+Now that we've done a walkthrough of the three tests, let's fix them.
+
+The first two tests assumed that our container container registry was already deployed to a resource group. However, we never deployed the `containerRegistry.bicep` file ourselves! The third test assumed we had an existing resource group to deploy to, but we never deployed that either! Let's go ahead and fix these assumptions now:
+
+1. Create a resource group in your subscription.
+1. Deploy the container registry bicep file to that resource group: `az deployment group create --resource-group <your-resource-group-name> --template-file .\containerRegistry.bicep` (You could also use the `Deploy-AzBicepFeature` helper here)
```

Comment:
Instead of using the `az` CLI, should this use the Azure PowerShell Module to be consistent?

Reply:
Also, do we need to do this step if the Pester test already spins up the resources?

Reply:
> Instead of using the `az` CLI, should this use the Azure PowerShell Module to be consistent?

Great idea! Changed over to New-AzResourceGroupDeployment cmdlet.



> Also, do we need to do this step if the Pester test already spins up the resources?

Resolved in other comments.